src/common/meta/Common.h

```diff
@@ -31,29 +31,30 @@ struct PartHosts {
 
 // ListenerHosts saves the listener type and the peers of the data replica
 struct ListenerHosts {
-  ListenerHosts(cpp2::ListenerType type, std::vector<HostAddr> peers)
-      : type_(std::move(type)), peers_(std::move(peers)) {}
+  ListenerHosts(PartitionID partId, std::vector<HostAddr> peers)
+      : partId_(partId), peers_(std::move(peers)) {}
 
   bool operator==(const ListenerHosts& rhs) const {
-    return this->type_ == rhs.type_ && this->peers_ == rhs.peers_;
+    return this->partId_ == rhs.partId_ && this->peers_ == rhs.peers_;
   }
 
   bool operator<(const ListenerHosts& rhs) const {
-    if (this->type_ == rhs.type_) {
+    if (this->partId_ == rhs.partId_) {
       return this->peers_ < rhs.peers_;
     }
-    return this->type_ < rhs.type_;
+    return this->partId_ < rhs.partId_;
   }
 
-  cpp2::ListenerType type_;
+  PartitionID partId_;
   // peers is the part peers which would send logs to the listener
   std::vector<HostAddr> peers_;
 };
 
 using PartsMap = std::unordered_map<GraphSpaceID, std::unordered_map<PartitionID, PartHosts>>;
 // ListenersMap is used for listener replica to get its peers of data replica
 using ListenersMap =
-    std::unordered_map<GraphSpaceID, std::unordered_map<PartitionID, std::vector<ListenerHosts>>>;
+    std::unordered_map<GraphSpaceID,
+                       std::unordered_map<cpp2::ListenerType, std::vector<ListenerHosts>>>;
```

Comment:
Previously `spaceId, partId` ->`mutile type ListenerHost`
Now s`paceId, type` ->`multi partId ListenerHosts`
Is the purpose of this change to make the search faster?

Reply:
Yes. Consider when we support multiple type of listener. Once we add a new type of listener by `add listener xxx`, we can't efficiently tell whether we already has a listener of type `xxx`, we need to iterate all partitions. And for each partitions, we need to iterate the vector `ListenerHosts` again.

It is a reason that we can't support multiple listener type in the same process.

Reply:
Good, Got it, thx